go.work

```diff
@@ -0,0 +1,6 @@
+go 1.20
+
+use (
+	.
+	./test
+)
```

Comment:
I think the Go workspace is to support the separate go.mod under the `test` directory. Why not just use 1 go.mod for the whole project?

Reply:
To avoid the interchaintest and it's dependencies as dependencies for the horcrux binary. This minimizes required deps to download for make install for docker and release, for example.

Reply:
This is not how Go modules work. The compiler builds a dependency graph and is smart enough to only download dependencies used for the binary. Because the `test` package is (rightfully) never used in horcrux's `main()`; `test`'s dependencies are ignored.

I deleted the go.work and test/go.mod andI tested on [a separate branch](https://github.com/strangelove-ventures/horcrux/pull/176).

As you can see, no interchaintest modules were downloaded. 

```
❯ make install
go: downloading github.com/cosmos/cosmos-sdk v0.47.3
go: downloading google.golang.org/grpc v1.55.0
go: downloading golang.org/x/crypto v0.8.0
go: downloading github.com/cosmos/gogoproto v1.4.10
go: downloading github.com/Jille/raft-grpc-transport v1.4.0
go: downloading github.com/hashicorp/raft v1.5.0
go: downloading github.com/ecies/go/v2 v2.0.6
go: downloading golang.org/x/net v0.9.0
go: downloading google.golang.org/genproto v0.0.0-20230306155012-7f2fa6fef1f4
go: downloading go.etcd.io/bbolt v1.3.7
go: downloading github.com/decred/dcrd/dcrec/secp256k1/v4 v4.2.0
go: downloading github.com/ethereum/go-ethereum v1.11.5
go: downloading github.com/hashicorp/go-hclog v1.5.0
go: downloading golang.org/x/exp v0.0.0-20230515195305-f3d0a9c9a5cc
go: downloading github.com/fatih/color v1.13.0
go: downloading cosmossdk.io/math v1.0.1
go: downloading github.com/hdevalence/ed25519consensus v0.1.0
go: downloading github.com/cosmos/cosmos-proto v1.0.0-beta.2
go: downloading filippo.io/edwards25519 v1.0.0
```

However, I'm fine with this decision. I think documentation pointing out this decision for a go.work is useful for people unfamiliar with the codebase, though.

On my branch, I ran into a weird `go mod tidy` error and punted on it further:
```
❯ go mod tidy
go: github.com/gogo/protobuf@v1.3.3: reading github.com/gogo/protobuf/go.mod at revision v1.3.3: unknown revision v1.3.3
```

But, in theory, only one go.mod should be fine and shouldn't impact performance or maintenance.

The one caveat is if horcrux `main()` and `test` share transitive deps. Then one go.mod will use [MVS](https://go.dev/ref/mod#minimal-version-selection) which may have unintended consequences, which may be the problem with `gogo/protobuf` package above.

Reply:
The protobuf is due to the replace statement in interchaintest. We would need to copy all of the replaces also which are currently only in the test/go.mod.

Good to know about the dependency graph. I thought we would have a performance impact [here](https://github.com/strangelove-ventures/horcrux/blob/andrew/interchaintest/docker/horcrux/native.Dockerfile#L7-L9) since it prefetches go deps for docker caching, but I could be wrong.

Reply:
Oh yes, it might fetch all in that case.

Reply:
But that step may not be necessary since `go build` downloads deps anyway. But we may be trying to optimize docker builds in some fashion.